src/AfriCC/EPP/Client.php

```diff
@@ -329,49 +329,55 @@ private function recv($length)
      * send data to socket
      * @param string $buffer
      */
-    private function send($buffer)
+    public function send($buffer)
     {
-        $info = stream_get_meta_data($this->socket);
-        $hard_time_limit = time() + $this->timeout + 2;
-        $length = mb_strlen($buffer, 'ASCII');
-
+        $info = array();
         $pos = 0;
-        while (!$info['timed_out'] && !feof($this->socket)) {
-            // Some servers don't like alot of data, so keep it small per chunk
-            $wlen = $length - $pos;
+        $length = null;
```

Comment:
I agree completely that this method might need a refactoring, but not really sure what your changes will do.